internal/metrics/tx_mapper_db.go

```diff
@@ -309,6 +311,76 @@ func (tm *TxMapperDB) AddValidatorRegistryEvent(ctx context.Context, vr *validat
 	return tx.Commit(ctx)
 }
 
+func (tm *TxMapperDB) UpdateValidatorStatus(ctx context.Context) error {
+	batchSize := 100
+	jumpBy := 0
+	numWorkers := 5
+	sem := make(chan struct{}, numWorkers)
+	var wg sync.WaitGroup
+
+	for {
+		// Query a batch of validator statuses
+		validatorStatus, err := tm.dbQuery.QueryValidatorStatuses(ctx, data.QueryValidatorStatusesParams{
+			Limit:  int32(batchSize),
+			Offset: int32(jumpBy),
+		})
+		if err != nil {
+			if err == pgx.ErrNoRows {
+				break
+			}
+			return err
+		}
+
+		if len(validatorStatus) == 0 {
+			break
+		}
+
+		// Launch goroutines to process each status concurrently
+		for _, vs := range validatorStatus {
+			sem <- struct{}{}
+			wg.Add(1)
+			go func(vs data.QueryValidatorStatusesRow) {
+				defer wg.Done()
+				defer func() { <-sem }()
+
+				validatorIndex := uint64(vs.ValidatorIndex.Int64)
+				//TODO: should we keep this log or remove it?
+				log.Debug().Uint64("validatorIndex", validatorIndex).Msg("validator status being updated")
+				validator, err := tm.beaconAPIClient.GetValidatorByIndex(ctx, "head", validatorIndex)
```

Comment:
we can optimize this call by batching the indices into the payload. the endpoint accepts an array of indices

Reply:
should we  maybe create an issue for this in the rolling-shutter repo to update beaconclient?